src/components/PopupWithForm.js

```diff
@@ -15,19 +14,46 @@ export default class PopupWithForm extends Popup {
     });
     return inputValues;
   }
-  // close() {
-  //   this._popupForm.reset();
-  //   super.close();
-  // }
+
+  open() {
+    super.open();
+  }
 
   setEventListeners() {
     //event listeners
-    this._popupForm.addEventListener("submit", (e) => {
+    this._popup = this._popupForm =
+      this._popupElement.querySelector(".modal__form");
+    this._popup.addEventListener("submit", (e) => {
       e.preventDefault();
       const inputValues = this._getInputValues();
+      console.log(inputValues);
       this._handleFormSubmit(inputValues);
+
+      this._popupForm.reset();
       this.close();
     });
     super.setEventListeners();
   }
+
+  // setSubmitAction(action) {
+  //   this._handleFormSubmit = action;
+  // }
+
+  renderLoading(isLoading) {
+    this._modalButton = this._popupElement.querySelector(".modal__button");
+    if (isLoading) {
+      this._modalButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```